src/providers.jl

```diff
@@ -1,17 +1,44 @@
+const valid_fftw_providers = let
+    # Hardcoded list of supported platforms
+    # In principle, we could check FFTW_jll.is_available() and MKL_jll.is_available()
+    # but then we would have to load MKL_jll which we want to avoid (lazy artifacts!)
+    platforms_providers = Dict(
+        Base.BinaryPlatforms.Platform("aarch64", "macos") => ("fftw",),
+        Base.BinaryPlatforms.Platform("aarch64", "linux"; libc = "glibc") => ("fftw",),
+        Base.BinaryPlatforms.Platform("aarch64", "linux"; libc = "musl") => ("fftw",),
+        Base.BinaryPlatforms.Platform("armv6l", "linux"; libc = "glibc", call_abi = "eabihf") => ("fftw",),
+        Base.BinaryPlatforms.Platform("armv6l", "linux"; libc = "musl", call_abi = "eabihf") => ("fftw",),
+        Base.BinaryPlatforms.Platform("armv7l", "linux"; libc = "glibc", call_abi = "eabihf") => ("fftw",),
+        Base.BinaryPlatforms.Platform("armv7l", "linux"; libc = "musl", call_abi = "eabihf") => ("fftw",),
+        Base.BinaryPlatforms.Platform("i686", "linux"; libc = "glibc") => ("fftw", "mkl"),
+        Base.BinaryPlatforms.Platform("i686", "linux"; libc = "musl") => ("fftw",),
+        Base.BinaryPlatforms.Platform("i686", "windows") => ("fftw", "mkl"),
+        Base.BinaryPlatforms.Platform("powerpc64le", "linux"; libc = "glibc") => ("fftw",),
+        Base.BinaryPlatforms.Platform("x86_64", "macos") => ("fftw", "mkl"),
```

Comment:
#280 shows that the hardcoded list a bad idea: there's no MKL 2024 build for this platform.

Reply:
But why not removing the automagic fallback, and only do what the user asks, and loudly and clearly fail if the artifact isn't available? In that case you can use is_available safely, since the user explicitly requested it

Reply:
Because the motivation for the PR and the problem I was trying to solve was that it failed. As a user I don't want to run into an error, I just want to use the provider that is available on my platform.

Reply:
See e.g. also https://julialang.slack.com/archives/C6A044SQH/p1701979919655409

Reply:
To clarify a bit more: In my case the MKL provider was chosen as the default for me by a package I was using, but in principle it could also be the other way around (that the currently hardcoded default FFTW_jll provider does not work but MKL_jll does). I think it would be less surprising and more user-friendly if the default provider would not be hard-coded to e.g. FFTW_jll but that it is checked which providers are actually available for the user and the default one is chosen as the first one among those. I don't think one should override explicit user settings but I think the default should work (and the error messages/list of valid providers should match the ones that are actually available).